R/zchunk_LA118.hydro.R

```diff
@@ -29,41 +29,170 @@ module_energy_LA118.hydro_DISABLED <- function(command, ...) {
-    L100.IEA_en_bal_ctry_hist <- get_data(all_data, "L100.IEA_en_bal_ctry_hist")
+
+    get_data(all_data, "temp-data-inject/L100.IEA_en_bal_ctry_hist") %>%
+      gather(year, value, -iso, -FLOW, -PRODUCT) %>%
+      mutate(year = as.integer(substr(year, 2, 5))) ->
+      L100.IEA_en_bal_ctry_hist
+
     A18.hydro_output <- get_data(all_data, "energy/A18.hydro_output")
 
     # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
+
+    # Calculation of economic hydropower potential by country, in EJ/yr
+    # Calculate a capacity factor for translating MW to GWh, using weighted average capacity factor of all existing dams
+    Hydropower_potential %>%
+      select(Installed_GWh, Installed_MW) %>%
+      summarise(Installed_GWh = sum(Installed_GWh), Installed_MW = sum(Installed_MW)) %>%
+      mutate(value = Installed_GWh / (Installed_MW * CONV_YEAR_HOURS * CONV_MIL_BIL)) %>%
+      .$value -> # Convert table to single number
+      Hydro_capfac
+
+    # Economic potential is what we are interested in from this database; however it is often not reported. Many countries without reported
+    # economic potential nevertheless have estimates of the technical potential.
+    # Calculate a translation from Technical potential to Economic potential, using weighted average among regions where both are reported
+    # First, for countries reporting in MW, convert to GWh (most countries have potentials as GWh but some are in MW)
+    Hydropower_potential %>%
+      mutate_if(is.integer, as.numeric) %>% # Convert columns that are getting read in as integers to numbers
+      mutate(Technical_GWh = replace(Technical_GWh, is.na(Technical_GWh), (Technical_MW * CONV_YEAR_HOURS * CONV_MIL_BIL * Hydro_capfac)[is.na(Technical_GWh)])) %>%
+      mutate(Economic_GWh = replace(Economic_GWh, is.na(Economic_GWh), (Economic_MW * CONV_YEAR_HOURS * CONV_MIL_BIL * Hydro_capfac)[is.na(Economic_GWh)])) ->
+      Hydropower_potential
+
+    # Among countries with both technical and economic potential reported, calculate an average translation from one to the other
+    Hydropower_potential %>%
+      summarise(value = sum(Economic_GWh[!is.na(Technical_GWh) & !is.na(Economic_GWh)]) / sum(Technical_GWh[!is.na(Technical_GWh) & !is.na(Economic_GWh)])) %>%
```

Comment:
Could this be redone as `filter(!is.na(Technical_GWh), !is.na(Economic_GWh)) %>% summarise(...)`? Seems like that would be a lot clearer.